README.md

```diff
@@ -13,175 +13,413 @@ Compartmentalization of host behavior hooks for JS.
+  // That object in turn has a null prototype.
+  // However some properties of `import.meta` are too expensive for a host
+  // to instantiate for every module, like Node.js's `import.meta.resolve`,
+  // which can be avoided for modules that never utter `import.meta`
+  // in their sources.
+  // This hook gets called for any module that utters `import.meta`,
+  // so some work can be deferred until just before the compartment
+  // initializes the module.
+  importMetaHook?: (fullSpecifier: string, importMeta: Object) => void,
 };
-// Exposed on global object
-// new Constructor per Compartment
-//
-// CreateRealm needs to be refactored to take params
-//  - intrinsics: an intrinsics record from
-//                6.1.7.4 Well-Known Intrinsic Objects
+
 interface Compartment {
-  constructor(
-    // extra bindings added to the global
-    endowments?: {
-      [globalPropertyName: string]: any
-    },
-    // need to figure out module attributes as it progresses
-    // maps child specifier to parent specifier
-    moduleMap?: {[key: FullSpecifier]: FullSpecifier | ModuleNamespace},
-    // including hooks like isDirectEvalHook
-    options?: CompartmentConstructorOptions
-  ): Compartment // an exotic compartment object
-
-  // access this compartment's global object, getter
-  globalThis: object;
-
-  // do an eval in this compartment
-  // default is strict indirect eval
-  evaluate(
-    // trusted types prep means use of `any`
-    src: any,
-    // FUTURE:
-    //   for other eval goals like Module, need to discuss import()/eval() to
-    //   get other Goals vs an option
-    // options?: object
-  ): any;
-
-  // Return signature differs to allow avoiding then() exports
-  // Used to ensure ability to be compatible with static import
-  async import(specifier: string): Promise<{namespace: ModuleNamespace}>;
-  // Desired by TC53
-  importNow(specifier: string): ModuleNamespace;
-  // Necessary to thread a module exports namespace from this compartment into
-  // the `moduleMap` Compartment constructor argument, without importing (and
-  // consequently executing) the module.
-  module(specifier: string): ModuleNamespace;
+  // Note: This single-argument form differs from earlier proposal versions,
+  // implementations of SES shim, and Moddable's XS, which accept three arguments,
+  // including a final options bag.
+  constructor(options?: CompartmentConstructorOptions): Compartment;
+
+  // Accessor for this compartment's globals.
+  // If inherit is true, globalThis is object identical to the incubating
+  // compartment's globalThis.
+  // If inherit is false, globalThis is a unique, ordinary object
+  // intrinsic to this compartment.
+  // The globalThis object initially has only "shared intrinsics",
+  // followed by compartment-specific "eval", "Function", and "Compartment",
+  // followed by any properties transferred from the "globals"
+  // constructor option with the semantics of Object.assign.
```

Comment:
We could easily omit the `globals` option entirely since it’s duplicative with `compartment.globalThis`, which allows the holder to do whatever they want.